kamu-core/src/domain/query_data_accessor.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use datafusion::prelude::SessionContext;
+use opendatafabric::{DataSlice, DatasetAlias};
+
+use super::InternalError;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub trait QueryDataAccessor: Send + Sync {
+    fn session_context(&self) -> Result<SessionContext, InternalError>;
+
+    fn dataset_object_store_url(&self) -> url::Url;
+
+    fn data_object_store_path(
+        &self,
+        dataset_alias: &DatasetAlias,
```

Comment:
this seems to have an assumption that datasets in a bucket are organized by aliases and not by IDs - we'd want to be flexible to support both

Reply:
I've switched the interface to `DatasetHandle`. But we always work with the already resolved datasets here, so both an ID and an alias are available